src/flexmeasures_client/client.py

```diff
@@ -0,0 +1,280 @@
+        - the client polling timed out (as indicated by the client's self.polling_timeout)
+        """  # noqa: E501
+        url = self.build_url(uri, path=path)
+        print(url)
+
+        headers = await self.get_headers(include_auth=include_auth)
+        self.start_session()
+
+        self.polling_step = 0
+        self.reauth_once = True  # reset this counter once when starting polling
+        try:
+            async with async_timeout.timeout(self.polling_timeout):
+                while self.polling_step < self.max_polling_steps:
+                    try:
+                        async with async_timeout.timeout(self.request_timeout):
+                            response = await self.request_once(
+                                method=method,
+                                url=url,
+                                params=params,
+                                headers=headers,
+                                json=json,
+                            )
+                            break
+                    except asyncio.TimeoutError:
+                        print(
+                            f"Client request timeout occurred while connecting to the API. Retrying in {self.polling_interval} seconds..."  # noqa: E501
+                        )
+                        self.polling_step += 1
+                        await asyncio.sleep(self.polling_interval)
+                    except (ClientError, socket.gaierror) as exception:
+                        raise ConnectionError(
+                            "Error occurred while communicating with the API."
+                        ) from exception
+        except asyncio.TimeoutError as exception:
+            raise ConnectionError(
+                "Client polling timeout while connection to the API."
+            ) from exception
+
+        check_content_type(response)
+
+        return cast(dict[str, Any], await response.json()), response.status
+
+    async def request_once(
+        self,
+        method: str,
+        url: str,
+        params: dict[str, Any] | None = None,
+        headers: dict | None = None,
+        json: dict | None = None,
+    ):
+        """Sends a single request to FlexMeasures and checks the response"""
+        response = await self.session.request(
+            method=method,
+            url=url,
+            params=params,
+            headers=headers,
+            json=json,
+            ssl=self.ssl,
+        )
+        await check_response(
+            self,
+            response,
+        )
+        print(response.headers)
+        return response
+
+    def start_session(self):
+        """If there is no session, start one"""
+        if self.session is None:
+            self.session = ClientSession()
+
+    async def get_headers(self, include_auth: bool) -> dict:
+        """If the request needs to be authenticated check if there is a access_token or request one. Then create the headers dict"""  # noqa: E501
```

Comment:
Maybe, we should move `If the request needs to be authenticated check if there is a access_token or request one.` to line 152 and say something like`Create HTTP headers dictionary with content type and, optionally, access token.`